include/envoy/event/dispatcher.h

```diff
@@ -153,6 +153,17 @@ class Dispatcher {
    */
   enum class RunType { Block, NonBlock };
   virtual void run(RunType type) PURE;
+
+  /**
+   * Optionally sets a custom buffer factory for the dispatcher.  This factory may be used by new
+   * connections on connection creation.
+   */
+  virtual void setBufferFactory(Buffer::FactoryPtr&& factory) PURE;
```

Comment:
Ah, I thought it'd be useful for consistency but I'm happy to remove it - with a bit of test cleanup I don't think it needs to be set at runtime.